app/apphandlers/processorAppHandler.go

```diff
@@ -88,81 +90,105 @@ func (processor *ProcessorApp) StartRudderCore(ctx context.Context, options *app
 	destinationdebugger.Setup(backendconfig.DefaultBackendConfig)
 
 	migrationMode := processor.App.Options().MigrationMode
-	//IMP NOTE: All the jobsdb setups must happen before migrator setup.
-	gatewayDB.Setup(jobsdb.Read, options.ClearDB, "gw", gwDBRetention, migrationMode, true, jobsdb.QueryFiltersT{})
-	defer gatewayDB.TearDown()
-
-	var routerDB jobsdb.HandleT = jobsdb.HandleT{}
+	reportingI := processor.App.Features().Reporting.GetReportingInstance()
 
-	var tenantRouterDB jobsdb.MultiTenantJobsDB = &jobsdb.MultiTenantLegacy{HandleT: &routerDB} //FIXME copy locks ?
+	//IMP NOTE: All the jobsdb setups must happen before migrator setup.
+	gwDB := jobsdb.NewForRead("gw", jobsdb.WithRetention(gwDBRetention), jobsdb.WithMigrationMode(migrationMode),
+		jobsdb.WithStatusHandler(), jobsdb.WithQueryFilterKeys(jobsdb.QueryFiltersT{}))
```

Comment:
[minor] for readability:

```suggestion
	gwDB := jobsdb.NewForRead(
	    "gw", 
	    jobsdb.WithRetention(gwDBRetention), 
	    jobsdb.WithMigrationMode(migrationMode),
	    jobsdb.WithStatusHandler(), 
	    jobsdb.WithQueryFilterKeys(jobsdb.QueryFiltersT{})
        )
```

Reply:
done